lang/java/avro/src/main/java/org/apache/avro/Schema.java

```diff
@@ -191,6 +206,57 @@ void setLogicalType(LogicalType logicalType) {
     this.logicalType = logicalType;
   }
 
+  /**
+   * For the Avro-defined primitive and complex data types this method returns
+   * matching AvroDatatype objects. The Avro-provided logical data types
+   * themselves implement the AvroDatatype interface already.
+   * 
+   * @return An object with more metadata about the data type and methods to
+   *         convert values
+   */
+  public AvroDatatype getDataType() {
+    if (logicalType != null && logicalType instanceof AvroDatatype) {
+      return (AvroDatatype) logicalType;
```

Comment:
So you can create new customized LogicalTypes and use them as long as they implement AvroDatatype.